src/daft-io/src/local.rs

```diff
@@ -84,49 +98,83 @@ pub struct LocalFile {
 #[async_trait]
 impl ObjectSource for LocalSource {
     async fn get(&self, uri: &str, range: Option<Range<usize>>) -> super::Result<GetResult> {
-        const TO_STRIP: &str = "file://";
-        if let Some(p) = uri.strip_prefix(TO_STRIP) {
-            let path = std::path::Path::new(p);
-            Ok(GetResult::File(LocalFile {
-                path: path.to_path_buf(),
-                range,
-            }))
-        } else {
-            return Err(Error::InvalidFilePath {
-                path: uri.to_string(),
-            }
-            .into());
-        }
+        const LOCAL_PROTOCOL: &str = "file://";
+        let path = uri.strip_prefix(LOCAL_PROTOCOL).unwrap_or(uri);
```

Comment:
I tried porting this to `Url::parse()`, but parsing relative URLs without a base isn't supported by the URL standard, currently.

Reply:
I also changed the semantics to not require the `"file://"` protocol prefix, since we found that requirement to be too strict for Ray Data users, and all other data libraries that I'm aware of accept local paths without such a prefix.

Reply:
discussed to revert.